### PR TITLE
Allow filtering by document type

### PR DESCRIPTION
**Pull request by peteglondon**
This commit allows filtering by document_type:
It is dependant on [This PR in content-performance-manager](https://github.com/alphagov/content-performance-manager/pull/964)

screen shot 2018-10-19 at 16 34 55

Url to test : http://content-data-admin.dev.gov.uk/content?utf8=%E2%9C%93&organisation_id=b8a6a8f2-d04f-4346-90c1-d28898565866&document_type=corporate_report&date_range=last-30-days&commit=Filter

### CODE REVIEW

**Comment by peteglondon** [comment]
[(2) Index page - filter by document type](https://trello.com/c/7zjqCznt/726-2-index-page-filter-by-document-type)

File: lib/gds_api/content_data_api.rb [hunk 1]
```diff
@@ -45,8 +49,14 @@ def time_series_request_url(base_path, from, to, metrics)
...
 
-  def content_items_url(from, to, organisation_id)
-    "#{content_data_api_endpoint}/content#{query_string(from: from, to: to, organisation_id: organisation_id)}"
+  def content_items_url(from, to, organisation_id, document_type)
+    params = {
+      from: from,
+      to: to,
+      organisation_id: organisation_id,
+      document_type: document_type
+ }.reject { |_, v| v.blank? }
```
Comment (theseanything) [inline lib/gds_api/content_data_api.rb]: Indent

File: spec/features/index_page_spec.rb [hunk 2]
```diff
@@ -103,4 +104,33 @@
...
+
+  context 'filter by document_type' do
+    before do
+      content_data_api_has_content_items(from: from, to: to, organisation_id: 'org-id', document_type: 'news_story', items: [items.first])
+      select 'News story', from: 'document_type'
+      click_on 'Filter'
+    end
+
+    it 'makes request to api with correct document_type' do
+      expect(page).to have_content('Content data')
```
Comment (theseanything) [inline spec/features/index_page_spec.rb]: This test seem to be concerned with implementation detail. Incorrect API calls should be caught by the tests below? Probably can remove this.

**Comment by pmanrubia** [comment]
@theseanything I have addressed your comments on top of @peteglondon's PR. Could you please check again?

File: spec/features/index_page_spec.rb [hunk 3]
```diff
@@ -103,4 +104,29 @@
...
+      click_on 'Filter'
+    end
+
+    it 'selects the document_type in the dropdown menu' do
+      expect(page).to have_select('document_type', selected: 'News story')
+    end
+
+    it 'renders the filtered results' do
+      table_rows = extract_table_content('.content-table table')
+      expect(table_rows.count).to eq(2)
```
Comment (theseanything) [inline spec/features/index_page_spec.rb]: Can we more explicity check that our items shown in the table are all of the correct `Content Type`?

Comment (theseanything) [reply spec/features/index_page_spec.rb]: Would like to only use one expect per example - use the `all` matcher we can remove counting the rows and make our test less fragile.
```
_header = table_rows.shift
expect(table_rows).to all(inlcude('News story'))
```

**Comment by pmanrubia** [comment]
@theseanything I have addressed your comments. Can you please have a look?

**Pull request merged**